sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp

```diff
@@ -91,6 +92,7 @@ ur_exp_command_buffer_handle_t_::~ur_exp_command_buffer_handle_t_() {
   if (WaitEvent) {
     WaitEvent->RefCount.decrementAndTest();
   }
+  Device->RefCount.decrementAndTest();
```

Comment:
While working on L0 memory leak, I observed that there is no cleanup process called after the command_buffer destructor to free memory when objects are no longer referenced (RefCount == 0).
This object must therefore be destroyed if RefCount == 0.

```suggestion
  urDeviceRelease(Device);
```

Reply:
Ah good point! I didn't realise the refcounts aren't actually used for devices (always 1) but only sub-devices. I've updated it to use both retain and release functions now!